addons/medical_damage/functions/fnc_getTypeOfDamage.sqf

```diff
@@ -17,24 +17,21 @@
 
 params ["_typeOfProjectile"];
 
-// --- projectiles
-if (_typeOfProjectile isKindOf "BulletBase") exitWith {"bullet"};
-if (_typeOfProjectile isKindOf "ShotgunBase") exitWith {"bullet"};
-if (_typeOfProjectile isKindOf "GrenadeCore") exitWith {"grenade"};
-if (_typeOfProjectile isKindOf "TimeBombCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "MineCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "FuelExplosion") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "ShellBase") exitWith {"shell"};
-if (_typeOfProjectile isKindOf "RocketBase") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "MissileBase") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "LaserBombCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "BombCore") exitWith {"explosive"};
-if (_typeOfProjectile isKindOf "Grenade") exitWith {"grenade"};
+private _damageType = GVAR(damageTypeCache) getVariable _typeOfProjectile;
 
-// --- non-projectiles reported by custom handleDamge wrapper
-if ((_typeOfProjectile select [0,1]) isEqualTo "#") then {
-    _typeOfProjectile = _typeOfProjectile select [1];
+if (isNil "_damageType") then {
+    _damageType = switch (true) do {
+        // -- non-projectiles reported by custom handleDamage wrapper
+        case ((_typeOfProjectile select [0,1]) isEqualTo "#"): { _typeOfProjectile select [1] };
+        // -- projectiles
+        case (isText (configFile >> "CfgAmmo" >> _typeOfProjectile >> "ACE_damageType")): { getText (configFile >> "CfgAmmo" >> _typeOfProjectile >> "ACE_damageType") };
+        default {
+            WARNING_1("Damage Type [%1] has no ACE_damageType",_typeOfProjectile);
+            toLower _typeOfProjectile
```

Comment:
```suggestion
            "bullet"
```

Reply:
Currently anything that isn't a [valid damage type](https://github.com/acemod/ACE3/blob/1f2004741027fcdfbc555f18c7a872d7df347d46/addons/medical_damage/functions/fnc_woundsHandlerSQF.sqf#L29-L32) uses `unknown` type (which _happens_ to be identical to `bullet`)  so I'd suggest returning that here to make the behaviour more explicit.

Reply:
I should have looked at this better before making these suggestions
the reason it passed `toLower _typeOfProjectile` is it may be passing calls from `ace_medical_fnc_addDamageToUnit` or anything else calling `[QEGVAR(medical,woundReceived)` event

so, we don't want WARNING if cfgAmmo doesn't exist, and we still want to pass _typeOfProjectile instead of a default value

Reply:
So it can be called with an already valid damage type, and will just return that?
We could pre-cache these values (e.g. "bullet" => "bullet") in `parseConfigForInjuries` so they'd skip this lookup entirely. In fact, that should allow us to remove the special case for things like `#collision`